src/network/networkd-dhcp4.c

```diff
@@ -1319,7 +1319,29 @@ int dhcp4_configure(Link *link) {
         if (r < 0)
                 return log_link_warning_errno(link, r, "DHCP4 CLIENT: Failed to set callback: %m");
 
-        r = sd_dhcp_client_set_request_broadcast(link->dhcp_client, link->network->dhcp_broadcast);
+        /* Some interfaces require that the DHCPOFFER message is being broadcast because
+         * they can't handle unicast messages while not fully configured.
+         * Setting device property ID_NET_NEEDS_DHCP_BROADCAST to "1" can be used to
+         * override the dhcp_broadcast flag for such interfaces. */
+        dhcp_broadcast = link->network->dhcp_broadcast;
+        if (dhcp_broadcast < 0) {
+                const char *needs_dhcp_broadcast = NULL;
+
+                r = sd_device_get_property_value(link->sd_device,
+                                                 "ID_NET_NEEDS_DHCP_BROADCAST",
+                                                 &needs_dhcp_broadcast);
+                if (r >= 0) {
```

Comment:
Not necessary to initialize `needs_dhcp_broadcast` (and it is not necessary the name is so explanative here).

Also, not a matter, but `r` is only used here, so,
```
const char *val;

if (sd_device_get_property_value(link->sd_device, "ID_NET_NEEDS_DHCP_BROADCAST", &val) >= 0) {
```